### PR TITLE
[FIX] web: code editor shouldn't accept markup as value

### PR DESCRIPTION
**Pull request by jpp-odoo**
Before this commit, when using the ace field widget with a html field, the markup value of the field was sent to the code editor. The issue with this is that the ace library only accept strings (it will extract the string from the markup), and when ask for the current value, it will respond in string. Because of this, the ace field will always be considered as dirty, even if not modified, because it will compare the initially sent value (a markup) with the value given by the library (a string).

Now, the formatText used on the ace field will always send a string (remove the markup). And the code editor will check the props to only accept strings.

### CODE REVIEW

**Comment by Iucapad** [comment]
Thanks for the fix that turns the AceField dirty 👍

File: addons/web/static/src/views/fields/formatters.js [hunk 1]
```diff
@@ -449,7 +449,18 @@ export function formatSelection(value, options = {}) {
  * @returns {string}
  */
 export function formatText(value) {
-    return value || "";
+    return value.toString() || "";
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/formatters.js:452]: value ? value.toString() : "";

File: addons/web/static/tests/views/fields/formatters_tests.js [hunk 2]
```diff
@@ -15,6 +16,7 @@ import {
     formatMonetary,
     formatPercentage,
     formatReference,
+    formatString,
```
Comment (aab-odoo) [inline addons/web/static/tests/views/fields/formatters_tests.js:19]: formatText

**Comment by jpp-odoo** [comment]
Hello @odoo/rd-security, can we have a review ? There are two new markups in the tests :

- [odoo/addons/web/static/tests/views/fields/formatters_tests.js:249]
```js
assert.strictEqual(formatText(markup("<p>This is a Test</p>")), "<p>This is a Test</p>");
```
This one tests that the markups are correctly removed when using `formatText`.


- [odoo/addons/web/static/tests/core/code_editor_tests.js:115]
```js
const textMarkup = markup<div>Some Text</div>;
```
This one tests that using a markup text in code_editor will raise a validation error.

Thanks

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: https://github.com/odoo/odoo/pull/129878#issuecomment-1655126693

**Pull request closed without merging**